test/topology/test_zero_token_node_nodetool.py

```diff
@@ -0,0 +1,128 @@
+import subprocess
+import asyncio
+from collections.abc import Coroutine
+from test.pylib.manager_client import ManagerClient
+import pytest
+
+
+@pytest.mark.asyncio
+async def validate_status_operation(result, live_eps, down_eps, leaving):
+    """Validates the nodetool status output"""
+
+    lines = result.split('\n')
+    i = 0
+    dc_line = lines[i]
+    assert dc_line.startswith("Datacenter:")
+    result_dc = dc_line.split()[1]
+    assert result_dc == "datacenter1"
+    dc_line_len = len(dc_line)
+
+    i += 1
+    assert lines[i] == "=" * dc_line_len
+
+    i += 1
+    assert lines[i] == "Status=Up/Down"
+
+    i += 1
+    assert lines[i] == "|/ State=Normal/Leaving/Joining/Moving"
+
+    i += 1
+    assert lines[i].split() == ["--", "Address", "Load", "Tokens", "Owns", "Host", "ID", "Rack"]
+
+    for j in range(len(live_eps + down_eps)):
+        i += 1
+        assert lines[i] != ""
+        status_state, ep = tuple(lines[i].split())[:2]
+
+        if ep in live_eps:
+            assert status_state[0] == 'U'
+        else:
+            assert status_state[0] == 'D'
+
+        if ep in leaving:
+            assert status_state[1] == 'L'
+        else:
+            assert status_state[1] == 'N'
+
+
+async def wait_for_first_completed(coros: list[Coroutine]):
+    done, pending = await asyncio.wait([asyncio.create_task(c) for c in coros], return_when=asyncio.FIRST_COMPLETED)
+    for t in pending:
+        t.cancel()
+    for t in done:
+        await t
+
+
+@pytest.mark.asyncio
+async def test_zero_token_node_normal(manager: ManagerClient):
+    await manager.servers_add(3)
+    servers = await manager.running_servers()
+
+    await manager.servers_add(servers_num=2, config={'join_ring': False})
+
+    exe_path = await manager.server_get_exe(servers[0].server_id)
+    cmd = [exe_path, "nodetool", "status"] + ["--logger-log-level",
+                                              "scylla-nodetool=trace",
+                                              "-h", servers[0].ip_addr]
+
+    result = subprocess.run(cmd, capture_output=True, text=True)
+
+    live_eps = await manager.api.client.get_json("/gossiper/endpoint/live", host=servers[0].ip_addr)
+    down_eps = await manager.api.client.get_json("/gossiper/endpoint/down", host=servers[0].ip_addr)
+    leaving = await manager.api.client.get_json("/storage_service/nodes/leaving", host=servers[0].ip_addr)
```

Comment:
In https://github.com/scylladb/scylladb/pull/20909#discussion_r1782563199, I asked for a test that verifies API endpoints when zero-token are in the cluster. The test you added doesn't do this. For example, `live_eps + down_eps + leaving` could be empty and the test would pass.

What we should do here is test that `live_eps`, `down_eps`, `leaving` are correct. In this test case, `live_eps` should contain IPs of all nodes (including zero-token nodes), each IP should appear once, and only nodes' IPs should appear. Also, `down_eps` and `leaving` should be empty. Other test cases should work similarly, just with different conditions.

My initial idea was that we don't run `nodetool status` in this test and we only test all API endpoints used by it (`test_status_with_zero_token_nodes` already tests `nodetool status` assuming API endpoints work correctly). This would require us to also check other API endpoints like `/storage_service/host_id`.

But now I think that what you did can also be fine if you improve it. If you check in `validate_status_operation` that host IDs are correct, then we will know that `/storage_service/host_id` works correctly for zero-token nodes. You should also check that `nodetool status` shows 0 tokens for zero-token nodes and a correct rack.

I also like what you did because it allows us to test one more thing easily -- running `nodetool status` on a zero-token node. It should work just like on a token-owning node. You can run `nodetool status` twice in this test case (once on a zero-token node) and test both runs in the same way. In other test cases it may be not that easy, but you can try.

Reply:
Done.

>In other test cases it may be not that easy, but you can try.
Not possible for other cases, since node is down